### DISCUSSION TITLE
change playback speed of an MP3/wav?

### DISCUSSION THREAD

**Discussion by kokospalme** [pschatzmann/arduino-audio-tools#499] [Q&A] [answered]
Hello folks,
I was wondering if there is a chance to change the playback speed (with/without pitch) of an MP3 or wav file?
This could be interesting for some music/DJ projects...
Thank you

**Comment by pschatzmann** [comment, marked as answer]
MP3 should be possible but WAV is much more efficient: To play the output speed with pitch change just change the output sample rate. You can also compbine this with [Pitch Shifting](https://github.com/pschatzmann/arduino-audio-tools/wiki/Pitch-Shifting) though I would tend to think that the quality will not be very good...

**Comment by kokospalme** [comment]
thank you for the quick  answer. Why do you think that the quality won't be good? What are the influencing factors?
And what do I have to do to change the playback speed without pitch? Is this even possible?

**Comment by pschatzmann** [comment]
Pitch shifting is generating additional waves or removing waves from the signal. I think this is quite tricky to do w/o distorting the original signal. I am providing different implementations, so you have something to play with.

I never tried to change the sample rate during playback, so I can't provide any comments for that...

**Comment by kokospalme** [comment]
okay, I'll try...

**Comment by pschatzmann** [comment]
I was rewriting the ResampleStream class to support a dynamic factor. I think this is the cleanest solution to change the speed and I would expect the quality to be acceptable. Here is an example: https://github.com/pschatzmann/arduino-audio-tools/blob/main/jupyter/TestsResample.ipynb
